File: datalib/src/main/java/nl/acidcats/tumblrlikes/core/models/tumblr/TumblrLikesResponse.java
package nl.acidcats.tumblrlikes.core.models.tumblr;

import com.google.auto.value.AutoValue;
import com.squareup.moshi.Json;
import com.squareup.moshi.JsonAdapter;
import com.squareup.moshi.Moshi;

import java.util.List;

/**
 * Created by stephan on 28/03/2017.
 */

@AutoValue
public abstract class TumblrLikesResponse {
    @Json(name = "liked_posts")
    public abstract List<TumblrLikeVO> likes();

    public static JsonAdapter<TumblrLikesResponse> jsonAdapter(Moshi moshi) {
        return new AutoValue_TumblrLikesResponse.MoshiJsonAdapter(moshi);
    }
}

File: datalib/src/main/kotlin/nl/acidcats/tumblrlikes/data_impl/likesdata/models/TumblrLikesResponse.kt
package nl.acidcats.tumblrlikes.data_impl.likesdata.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Created on 24/10/2018.
 */
@JsonClass(generateAdapter = true)
data class TumblrLikesResponse(
        @Json(name = "liked_posts")
        val likes: List<TumblrLikeVO>
)